test/unit/queue/src/QueueTest.cpp

```diff
@@ -26,6 +26,28 @@ using TestQueues = alpaka::meta::Concatenate<
 #endif
     >;
 
+//! Equivalent too CHECK but conditions with false negatives
+//!
+//! This is required when checking if an event or a queue is finished/empty.
+//! \warning This macro can be used on host only!
+//!
+//! \param count number of checks until the check will use the last result of the condition
```

Comment:
In the implementation it only applies for negative results. The first positive result will be used.